mmeds/util.py

```diff
@@ -1583,3 +1600,86 @@ def get_subject_type(subject_file):
         if not sub_type == subject_type:
             subject_type = 'mixed'
     return subject_type.lower()
+
+
+def get_file_index_entry_location(path, tool, entry, testing=False):
+    """ Get entry from a non-active Tool's file_index.tsv file, for use in cross-tool analysis """
```

Comment:
what is cross-tool analysis and why don't you know where the file_index you want is located?

Reply:
so this is what I came up with for when we need to use (for example) a qiime2 feature table in the lefse tool. The path to that feature table is stored in the Qiime2_# folder's `file_index.tsv`, but knowing where that is is a little complicated. To get it another way we would have to store every Tool object ever run in perpetuity in order to pass it as an argument at this stage. So what we do here is look at the most recent Qiime2 folder for the study (i.e. if there's a Qiime2_1 and Qiime2_0 look at _1 first) and go down the list looking for the first occurrence of an existing ~whatever file it is~. So we do know where all the file_index are, but we need to check them in a particular order

Reply:
hmm I see, this is a fine solution to get functionality working but it feels like there should be a better solution possible if not in the current framework. Seems like this problem would need to be solved at a higher level. I think we should make a note to discuss file pathing in mmeds sometime in the near future.

Reply:
i'll make an issue